File: problems/2364-count-number-of-bad-pairs/solution.py
# Link: https://leetcode.cn/problems/count-number-of-bad-pairs/
# Level: Medium
# Date: 2025-04-18

class Solution:
    def countBadPairs(self, nums: List[int]) -> int:
        n = len(nums)
        cnt = Counter()
        ans = 0
        for i, num in enumerate(nums):
            ans += i - cnt[num - i]
            cnt[num - i] += 1
        return ans

File: problems/2364-count-number-of-bad-pairs/solution.cpp
// Link: https://leetcode.cn/problems/count-number-of-bad-pairs/
// Level: Medium
// Date: 2025-04-18

class Solution {
    public:
        long long countBadPairs(vector<int>& nums) {
            long long ans = 0;
            int n = nums.size();
            unordered_map<int, int> mp;
            for (int i = 0; i < n; i++) {
                ans += i - mp[nums[i] - i];
                mp[nums[i] - i]++;
            }
            return ans;
        }
};
